src/serial/serial.c

```diff
@@ -186,16 +186,31 @@ char serial_get_c(struct Serial *s)
         return serial_get_c_wait(s, &c, portMAX_DELAY) ? c : 0;
 }
 
-int serial_get_line_wait(struct Serial *s, char *l, const size_t len,
+/**
+ * Reads in a line from a serial device delimeted by \n.  The data is
+ * written to buff BUT MAY NOT BE NULL TERMINATED.  NULL termination is the
```

Comment:
Because I need to sometimes know what the NULL terminator was.  I need to handle it differently if its a `\r\n` versus a `\0` (like you can see on some netcat commands).  By not modifying the serial here, and instead handling this higher up, I gain more flexibility in my decision making process.  Plus this simplifies the serial code more by making it do less (which is usually a good practice where reasonable).